src/CheckOut.js

```diff
@@ -291,6 +297,24 @@ class CheckOut extends React.Component {
     }
   }
 
+  showBlockModal = (patron) => {
+    const {
+      resources,
+    } = this.props;
+
+    const selManualPatronBlocks = get(resources, ['manualPatronBlocks', 'records'], []);
+    let manualPatronBlocks = selManualPatronBlocks.filter(p => p.borrowing === true);
+    manualPatronBlocks = manualPatronBlocks.filter(
+      p => moment(moment(p.expirationDate).format()).isSameOrAfter(moment().format())
+    );
+
+    const selAutomatedPatronBlocks = get(resources, ['automatedPatronBlocks', 'records'], []);
+    const automatedPatronBlocks = selAutomatedPatronBlocks.filter(p => p.blockBorrowing === true);
+
+    return (manualPatronBlocks.length > 0 && manualPatronBlocks[0].userId === patron.id)
```

Comment:
You've  made lots of refactoring, great job! I guess it would be also good to divide with commits refactoring with realization of ticket. Single commit makes git history cleaner but code reviewing more difficult.

Reply:
Fixed